public_html/wp-content/plugins/pattern-creator/src/index.js

```diff
@@ -24,13 +25,16 @@ apiFetch.use( filterEndpoints );
  * an unhandled error occurs, replacing previously mounted editor element using
  * an initial state from prior to the crash.
  *
- * @param {Element} target   DOM node in which editor is rendered.
- * @param {?Object} settings Editor settings object.
+ * @param {Element} target          DOM node in which editor is rendered.
+ * @param {Object}  settings        Editor settings.
+ * @param {number}  settings.postId ID of the current post.
  */
-export function reinitializeEditor( target, settings ) {
+export function reinitializeEditor( target, { postId, ...settings } ) {
 	unmountComponentAtNode( target );
 	const reboot = reinitializeEditor.bind( null, target, settings );
-	render( <Editor initialSettings={ settings } onError={ reboot } />, target );
+
+	dispatch( patternStore ).updateSettings( settings );
```

Comment:
Why do we need to `dispatch` here? Does the order matter?

Reply:
It's been removed from `<Editor />`, so this should only run on the initialization of the editor (or re-initialization if necessary), and since it's a synchronous call, it happens before rendering, and sets up the settings without needing a `useEffect`. 

See https://github.com/WordPress/gutenberg/blob/a233dc45f1316ab05764840eebcb94373400add4/packages/edit-site/src/index.js#L57-L58. I'll add a comment explaining this too.